core/src/types.rs

```diff
@@ -174,6 +174,41 @@ pub enum Origin {
 	Other(String),
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum NetworkMode {
+	/// Use both P2P and RPC for data retrieval
+	Both,
+	/// Use only P2P for data retrieval (RPC disabled)
+	P2POnly,
+	/// Use only RPC for data retrieval (P2P disabled)
+	RPCOnly,
+}
+
+impl clap::ValueEnum for NetworkMode {
+	fn value_variants<'a>() -> &'a [Self] {
+		&[Self::Both, Self::P2POnly, Self::RPCOnly]
+	}
+
+	fn to_possible_value(&self) -> Option<clap::builder::PossibleValue> {
+		Some(match self {
+			Self::Both => clap::builder::PossibleValue::new("both"),
+			Self::P2POnly => clap::builder::PossibleValue::new("p2p_only"),
+			Self::RPCOnly => clap::builder::PossibleValue::new("rpc_only"),
+		})
+	}
+}
+
+impl std::fmt::Display for NetworkMode {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		match self {
+			NetworkMode::Both => write!(f, "both"),
+			NetworkMode::P2POnly => write!(f, "p2p_only"),
+			NetworkMode::RPCOnly => write!(f, "rpc_only"),
+		}
+	}
+}
+
```

Comment:
Not sure if I'm missing something, but this should be possible with something like:

```rust
use clap::ValueEnum;

#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, ValueEnum/)]
#[serde(rename_all = "snake_case")]
enum NetworkMode {
    #[value(name = "both")]
    Both,
    #[value(name = "p2p_only")]
    P2POnly,
    #[value(name = "rpc_only")]
    RPCOnly,
}
```